Editor/Modules/TextureAnalyzer.cs

```diff
@@ -160,6 +176,37 @@ public IEnumerable<ProjectIssue> Analyze(ProjectAuditorParams projectAuditorPara
                 yield return ProjectIssue.Create(IssueCategory.AssetDiagnostic, k_TextureStreamingMipMapEnabledDescriptor, textureName)
                     .WithLocation(textureImporter.assetPath);
             }
+
+            if (ScanSolidTexture.IsTextureSolidColorTooBig(textureImporter, texture))
+            {
+                yield return ProjectIssue.Create(IssueCategory.AssetDiagnostic, k_TextureSolidColorDescriptor, textureName)
+                    .WithLocation(textureImporter.assetPath);
+            }
+        }
+
+        internal static void ResizeSolideTexture(string path)
+        {
+            var textureImporter = AssetImporter.GetAtPath(path) as TextureImporter;
+            if (textureImporter != null)
+            {
+                var originalValue = textureImporter.isReadable;
+                textureImporter.isReadable = true;
+                textureImporter.SaveAndReimport();
+
+                var texture = AssetDatabase.LoadAssetAtPath<Texture>(path) as Texture2D;
+                var color = texture.GetPixel(0, 0);
+                //Create a new texture as we can't resize the current one
+                var newTexture = new Texture2D(1, 1, TextureFormat.RGBA32, false);
```

Comment:
I am actually not 100% sure if there is a problem - just a thought:

I was thinking that instead of TextureFormat.RGBA32 we could also use the original texture's format "texture.format", just in case it was a rather "exotic" format.

Then there is one thought related to preserving the color:  If the format was for example BGRA32 for some strange reason (I am not sure who would use this :D), does the GetPixel() and SetPixel() preserve the old color or may do the mistake of swapping the red/green/blue.

Reply:
That is a good comment. However, I need to keep the RGBA32 format to prevent issue and be able to use SetPixel or EncodeToPNG. In other cases it will break

Reply:
Yes, it is probably ok!

I actually tried it on Android, e.g. to force a "bad format". It doesn't even seem possible for a artist for example, you probably would need to we write code to change the TextureFormat and save/re-import.

So let's ignore this until anyone reports an edge case, some problem!